src/main.cpp

```diff
@@ -299,7 +304,14 @@ void nimble_port_ll_task_func(void* args) {
 }
 }
 
-int main(void) {
+/**
+ * This is for fooling LTO to not remove a function referenced in inline asm
+ */
+static __attribute__((used)) void TaskSwitchDummy() {
+  vTaskSwitchContext();
+}
```

Comment:
This solution looks like a dirty hack, I would suggest to do the following more clear hack instead:
```diff
port.c:
void xPortPendSVHandler( void )
{
...
-    "   bl vTaskSwitchContext               \n"
+    "   bl _vTaskSwitchContext              \n"
...
}

+void __attribute__((used)) _vTaskSwitchContext(void)
+{
+    vTaskSwitchContext();
+}
```
Note: I found that this solution shouldn't add any overhead as `vTaskSwitchContext` will be inlined into `_vTaskSwitchContext`.